docs/guide/SecurityBC.md

```diff
@@ -9,6 +9,37 @@ The security design:
 1. is compatible with future Infrastructure-as-Code (IaC) deployments
 1. provides activity logging that can be used in an audit
 
+## RBAC Implementation details
+1. Users are assigned one or more roles. A user may 'take on' multiple roles, subject to defined rules.
+1. Roles are assigned Permissions
+1. The Identity and Access Proxy (Ory Oathkeeper) enforces access to endpoints based on permission.
+1. Backend API can optionally check permissions through Keto API.
+1. Mutually exclusive permission sets can be defined in the system to enforce seperation of duties. 
+
+## Enforcing Maker-Checker
+There are two approaches that can be taken to enforce a maker-checker validation flow.
+1. Enforce using roles and mutually exclusive permissions via security policies. I.e. Makers cannot also be checkers
+1. Enforce in application layer security rules. I.e. Makers cannot also be checkers within the same validation process. This is often implemented in the application layer when assigning makers and/or checkers as defined in a process flow, and by enforcing that a checker cannot be the same person as the maker in a validation flow. I.e. the code that enforces this will exist within each bounded context.

```

Comment:
```suggestion
1. Enforce in application layer security rules; i.e. makers cannot also be checkers within the same validation process. This is often implemented in the application layer when assigning makers and/or checkers as defined in a process flow, and by enforcing that a checker cannot be the same person as the maker in a validation flow. I.e. the code that enforces this will exist within each bounded context.

```